### PR TITLE
feat: (Day) Adds custom player name colors for in-party/in-guild/hostile

### PR DESCRIPTION
**Pull request by AlexVild**
Default hostile color:
image

Default party color:
image

Default guild color:
image

### CODE REVIEW

File: Intersect (Core)/CustomColors.cs [hunk 1]
```diff
@@ -82,6 +82,9 @@ public sealed partial class NamesNamespace
             public Dictionary<string, LabelColor> Players = new Dictionary<string, LabelColor>()
             {
                 {"Normal", new LabelColor(new Color(255, 255, 255, 255), Color.Black, new Color(180, 0, 0, 0))},
+                {"PartyMember", new LabelColor(new Color(255, 70, 192, 219), Color.Black, new Color(180, 0, 0, 0))},
+                {"GuildMember", new LabelColor(new Color(255, 5, 179, 26), Color.Black, new Color(180, 0, 0, 0))},
```
Comment (lodicolo) [inline Intersect (Core)/CustomColors.cs:86]: can you change these to just Party and Guild

Comment (lodicolo) [reply Intersect (Core)/CustomColors.cs:86]: reason: I want to take advantage of `nameof()` wherever possible in the future and I don't see us ever creating types called `PartyMember` and `GuildMember`

File: Intersect.Client/Entities/Player.cs [hunk 2]
```diff
@@ -2406,9 +2409,45 @@ public override void DrawName(Color textColor, Color borderColor, Color backgrou
...
+                // Guildies
+                else if (Globals.Me.IsInGuild && Guild == Globals.Me.Guild && CustomColors.Names.Players.TryGetValue("GuildMember", out var guildColors))
+                {
+                    textColor = guildColors.Name;
+                    borderColor = guildColors.Outline;
+                    backgroundColor = guildColors.Background;
+                }
+
+                // Enemies in PvP
+                if (!Globals.Me.IsAllyOf(this) && Globals.Me.MapInstance.ZoneType != MapZone.Safe && CustomColors.Names.Players.TryGetValue("Hostile", out var hostileColors))
```
Comment (lodicolo) [inline Intersect.Client/Entities/Player.cs:2430]: Is there a reason this isn't `else if`d?

File: Intersect.Client/Entities/Player.cs [hunk 3]
```diff
@@ -2406,9 +2409,45 @@ public override void DrawName(Color textColor, Color borderColor, Color backgrou
...
         }
 
+        public override bool IsAllyOf(Player en)
+        {
+            if (base.IsAllyOf(en)) 
+            {
+                return true;
+            }
+
+            return Globals.Me.IsInMyParty(en) || Globals.Me.IsInMyGuild(en) || en.MapInstance.ZoneType == MapZone.Safe;
```
Comment (lodicolo) [inline Intersect.Client/Entities/Player.cs:2448]: Shouldn't the first two conditions be `IsInMyParty()` and `IsInMyGuild()` without the `Globals.Me.` qualification?

The `IsAllyOf()` method is being misused if the `Globals.Me.` qualification.

File: Intersect.Client/Entities/Entity.cs [hunk 4]
```diff
@@ -489,6 +489,34 @@ public void AddAnimations(List<AnimationBase> anims)
...
+            }
+
+            // Events ONLY have allies
+            if (Type == EntityType.Event)
+            {
+                return true;
+            }
+
+            // Yourself is always an ally
+            if (en.Id == Globals.Me.Id)
```
Comment (lodicolo) [inline Intersect.Client/Entities/Entity.cs:512]: I think this method should be comparing `en.Id` to `Id` not `Globals.Me.Id`. The `IsAllyOf()` method is being misused if the `Globals.Me.` qualification.

**Review by lodicolo — CHANGES_REQUESTED** [review]

File: Intersect.Client/Entities/Player.cs (hunk 2 above)
Comment (AlexVild) [reply Intersect.Client/Entities/Player.cs:2430]: My reasoning is that in the _future_, if there are additional rules about what constitutes an ally (i.e, maybe you can kill guildies in an Arena map), then we won't miss this. Basically, if someone is hostile, their name color will _always_ be overridden.

File: Intersect.Client/Entities/Entity.cs (hunk 4 above)
Comment (AlexVild) [reply Intersect.Client/Entities/Entity.cs]: Good catch, fixed

File: Intersect (Core)/CustomColors.cs (hunk 1 above)
Comment (AlexVild) [reply Intersect (Core)/CustomColors.cs]: Good catch, fixed

File: Intersect.Client/Entities/Player.cs (hunk 3 above)
Comment (AlexVild) [reply Intersect.Client/Entities/Player.cs]: Good catch, fixed

**Review by lodicolo — APPROVED** [review]

**Pull request merged**